src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3359,48 +3421,55 @@ private bool HasTopLevelNullabilityConversion(TypeSymbolWithAnnotations source,
                     }
                     break;
 
+                case ConversionKind.ImplicitThrow:
+                    break;
+
                 case ConversionKind.Unboxing:
-                    if (!operandType.IsNull && targetType.IsTypeParameter())
+                    if (!operandType.IsNull)
```

Comment:
>operandType.IsNull [](start = 25, length = 18)

Can ```operandType.IsNull``` ever be true when the target type is nullable value type? Consider also setting result to nullable for such case.